keras_cv/models/backbones/mobilenet_v3/mobilenet_v3_backbone.py

```diff
@@ -0,0 +1,451 @@
+
+import copy
+
+from tensorflow import keras
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+from tensorflow.keras.utils import custom_object_scope
+
+from keras_cv import layers as cv_layers
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.mobilenet_v3.mobilenet_v3_backbone_presets import (  # noqa: E501
+    backbone_presets,
+)
+from keras_cv.utils.python_utils import classproperty
+
+CHANNEL_AXIS = -1
+BN_EPSILON = 1e-3
+BN_MOMENTUM = 0.999
+
+
+def adjust_channels(x, divisor=8, min_value=None):
+    """Ensure that all layers have a channel number divisible by the `divisor`.
+
+    Args:
+        x: integer, input value.
+        divisor: integer, the value by which a channel number should be
+            divisible, defaults to 8.
+        min_value: float, optional minimum value for the new tensor. If None,
+            defaults to value of divisor.
+
+    Returns:
+        the updated input scalar.
+    """
+
+    if min_value is None:
+        min_value = divisor
+
+    new_x = max(min_value, int(x + divisor / 2) // divisor * divisor)
+
+    # make sure that round down does not go down by more than 10%.
+    if new_x < 0.9 * x:
+        new_x += divisor
+    return new_x
+
+
+def apply_hard_sigmoid(x):
+    """Applies the hard sigmoid activation to x and returns the result.
+
+    Args:
+        x: input tensor
+    Returns:
+        the updated input tensor.
+    """
+
+    activation = layers.ReLU(6.0)
+
+    return activation(x + 3.0) * (1.0 / 6.0)
+
+
+def apply_hard_swish(x):
+    """Applies the hard swish activation to x and returns the result.
+
+    Args:
+        x: input tensor
+
+    Returns:
+        the updated input tensor.
+    """
+
+    return layers.Multiply()([x, apply_hard_sigmoid(x)])
+
+
+def apply_inverted_res_block(
+    x,
+    expansion,
+    filters,
+    kernel_size,
+    stride,
+    se_ratio,
+    activation,
+    expansion_index,
+):
+    """An Inverted Residual Block.
+
+    Args:
+        x: input tensor.
+        expansion: integer, the expansion ratio, multiplied with infilters to
+            get the minimum value passed to adjust_channels.
+        filters: integer, number of filters for convolution layer.
+        kernel_size: integer, the kernel size for DepthWise Convolutions.
+        stride: integer, the stride length for DepthWise Convolutions.
+        se_ratio: float, ratio for bottleneck filters. Number of bottleneck
+            filters = filters * se_ratio.
+        activation: the activation layer to use.
+        expansion_index: integer, a unique identification if you want to use
+            expanded convolutions. If greater than 0, an additional Conv+BN
+            layer is added after the expanded convolutional layer.
+
+    Returns:
+        the updated input tensor.
+    """
+    if activation == "hard_swish":
+        activation = apply_hard_swish
+    if activation == "relu":
```

Comment:
Thanks, looking at the code I now see why the activation cannot be a string. In this case the block should read: 
```python
if isinstance(activation, str):
    if activation == "hard_swish":
        activation = apply_hard_swish
    else:
        activation = keras.activations.get(activation)
```

This block will throw with an information error message if the string is not supported.